src/js/messaging/components/FolderNav.jsx

```diff
@@ -43,10 +127,13 @@ FolderNav.propTypes = {
       /* eslint-disable */
       folder_id: React.PropTypes.number.isRequired,
```

Comment:
When we were using the mock Heroku API, the 'X-Key-Inflection' header didn't work, so I just went with the snake cased keys. I just tried the 'X-Key-Inflection' again on the staging API request, and it seemed to work, so I can change this in a separate branch.